quickstarts/microsoft.databasewatcher/watcher-new-datastore/azuredeploy.json

```diff
@@ -0,0 +1,383 @@
+                    "$schema": "https://schema.management.azure.com/schemas/2019-04-01/deploymentTemplate.json#",
+                    "contentVersion": "1.0.0.0",
+                    "parameters": {
+                        "location": {
+                            "type": "string"
+                        },
+                        "clusterName": {
+                            "type": "string"
+                        },
+                        "principalId": {
+                            "type": "string"
+                        },
+                        "tenantId": {
+                            "type": "string"
+                        },
+                        "databaseName": {
+                            "type": "string"
+                        }
+                    },
+                    "resources": [
+                        {
+                            "type": "Microsoft.Kusto/clusters/databases",
+                            "apiVersion": "2023-05-02",
+                            "name": "[concat(parameters('clusterName'), '/', parameters('databaseName'))]",
+                            "location": "[parameters('location')]",
+                            "kind": "ReadWrite",
+                            "properties": {},
+                            "resources": [
+                                {
+                                    "type": "Microsoft.Kusto/Clusters/Databases/PrincipalAssignments",
+                                    "apiVersion": "2023-05-02",
+                                    "name": "[concat(parameters('clusterName'), '/', parameters('databaseName'), '/', guid(resourceGroup().id))]",
+                                    "dependsOn": [
+                                        "[parameters('databaseName')]"
+                                    ],
+                                    "properties": {
+                                        "tenantId": "[parameters('tenantId')]",
+                                        "principalId": "[parameters('principalId')]",
+                                        "role": "Admin",
+                                        "principalType": "App"
+                                    }
+                                }
+                            ]
+                        }
+                    ]
+                }
+            }
+        },
+        {
+            "copy": {
+                "name": "targetcopy",
+                "count": "[length(range(0, parameters('targetCount')))]"
+            },
+            "type": "Microsoft.DatabaseWatcher/watchers/targets",
+            "apiVersion": "[parameters('databaseWatcherApiVersion')]",
+            "name": "[concat(parameters('name'), '/', guid(resourceGroup().id, parameters('name'), string(copyIndex('targetcopy'))))]",
+            "dependsOn": [
+                "[parameters('name')]"
+            ],
+            "properties": "[json(parameters('targetProperties')[copyIndex('targetcopy')])]"
```

Comment:
Why is the whole properties body being passed as a parameter value? The individual properties should be declared and then the values should be parameterized. If you want you can use an `object` parameter type if you only want one parameter, but as it is currently authored this is an anti-pattern that will be confusing for any consumer of this template.

Reply:
Fixed